src/bindings/PyDP/mechanisms/mechanism.cpp

```diff
@@ -1 +1,202 @@
+          .def("NoisedValueAboveThreshold", &dp::GaussianMechanism::NoisedValueAboveThreshold)
+          .def("NoiseConfidenceInterval", py::overload_cast<double, double, double>(&dp::GaussianMechanism::<dp::ConfidenceInterval>NoiseConfidenceInterval),
+            py::arg("confidence_level"), py::arg("privacy_budget"), py::arg("noised_result"))
+          .def("Serialize", &dp::serialization::GaussianMechanism::Serialize)
+          .def("MemoryUsed", &dp::GaussianMechanism::MemoryUsed)
+          .def("CalculateStddev", &dp::GaussianMechanism::CalculateStddev)
+          .def("GetDelta", &dp::GaussianMechanism::GetDelta)
+          .def("GetL2Sensitivity", &dp::GaussianMechanism::GetL2Sensitivity)
+          .def("GetVariance", &dp::GaussianMechanism::GetVariance)
+          .def("CalculateDelta", &dp::GaussianMechanism::CalculateDelta)
+          .def("AddDoubleNoise", &dp::GaussianMechanism::AddDoubleNoise)
+          .def("AddInt64Noise", &dp::GaussianMechanism::AddInt64Noise)
+          .def_property_readonly("l2_sensitivity", &dp::GaussianMechanism::l2_sensitivity)
+          .def_property_readonly("delta", &dp::GaussianMechanism::delta)
+          .def_property_readonly("distro", std::unique_ptr<internal::dp::GaussianDistribution> distro)
+          .def("StandardNormalDistributionCDF", &dp::GaussianMechanism::StandardNormalDistributionCDF);
+  }
+};
+
+class declareMinVarianceMechanismBuilder {
+  public:
+    /* some work around here */
+    std::unique_ptr<dp::NumericalMechanism> build(double l2_sensitivity) {
+    dp::NumericalMechanismBuilder::Builder builder;
+    builder.SetL2Sensitivity(l2_sensitivity);
+    return std::move(builder.Build().value());
+  };
+
+    std::unique_ptr<dp::NumericalMechanismBuilder> clone() {
+    dp::NumericalMechanismBuilder cloner;
+    return std::move(cloner.Clone().value());
+  };
+
+    void declareMinVarianceMechanismBuilder(py::module &m) {
+      py::class_<dp::MinVarianceMechanismBuilder, dp::NumericalMechanismBuilder> min_varmech(m, "MinVarianceMechanismBuilder");
+      min_varmech.attr("__module__") = "pydp";
+      min_varmech
+          .def("GetMechanismFromBuilder", std::unique_ptr<dp::NumericalMechanism>::&dp::GaussianMechanism::GetMechanismFromBuilder);
+  }
+};
+
+void init_mechanisms_mechanism(py::module& m) {
+  declareNumericalMechanism num_obj = declareNumericalMechanism();
+  num_obj.declareNumericalMechanism(m);
+  declareNumericalMechanismBuilder num_build_obj = declareNumericalMechanismBuilder();
+  num_build_obj.declareNumericalMechanismBuilder(m);
+  declareLaplaceMechanism laplace_obj = declareLaplaceMechanism();
+  laplace_obj.declareLaplaceMechanism(m);
+  declareGaussianMechanism gaussian_obj = declareGaussianMechanism();
+  gaussian_obj.declareGaussianMechanism(m);
+  declareMinVarianceMechanismBuilder minvar_obj = declareMinVarianceMechanismBuilder();
+  minvar_obj.declareMinVarianceMechanismBuilder(m);
```

Comment:
What is the purpose of this mechanism?

Reply:
returns the mechanism with minimum variance for given parameters.  It chooses between Gaussian and Laplace mechanism.